packages/identity/src/identity/der.test.ts

```diff
@@ -0,0 +1,163 @@
+  test('Length of up to 127', () => {
+    const min = encodeLen(buf, 0, 1);
+    const max = encodeLen(buf, 0, 127);
+    expect(min).toEqual(1);
+    expect(max).toEqual(1);
+  });
+  test('Length of up to 255', () => {
+    const min = encodeLen(buf, 0, 128);
+    const max = encodeLen(buf, 0, 255);
+    expect(min).toEqual(2);
+    expect(max).toEqual(2);
+  });
+  test('Length of up to 65535', () => {
+    const min = encodeLen(buf, 0, 256);
+    const max = encodeLen(buf, 0, 65535);
+    expect(min).toEqual(3);
+    expect(max).toEqual(3);
+  });
+  test('Length of up to 16777215', () => {
+    const min = encodeLen(buf, 0, 65536);
+    const max = encodeLen(buf, 0, 16777215);
+    expect(min).toEqual(4);
+    expect(max).toEqual(4);
+  });
+  test('Length of up to 16777215', () => {
+    const min = encodeLen(buf, 0, 65536);
+    const max = encodeLen(buf, 0, 16777215);
+    expect(min).toEqual(4);
+    expect(max).toEqual(4);
+  });
+  test('Over maximum length', () => {
+    const shouldFail = () => encodeLen(buf, 0, 16777216);
+    expect(shouldFail).toThrowError('Length too long (> 4 bytes)');
+  });
+});
+
+describe('DecodeLenBytes', () => {
+  test('encoded length of zero', () => {
+    // length of buf doesn't matter
+    const buf = Uint8Array.from(new Array(2));
+    // 129 signifies a DER length of 2
+    buf[0] = 128;
+    const shouldFail = () => decodeLenBytes(buf, 0);
+    expect(shouldFail).toThrowError('Invalid length 0');
+  });
+  test('encoded length of 1', () => {
+    const buf = Uint8Array.of(100);
+    expect(decodeLenBytes(buf, 0)).toBe(1);
+  });
+  test('encoded length of 2', () => {
+    const buf = Uint8Array.from(new Array(2));
+    // 129 signifies a DER length of 2
+    buf[0] = 129;
+    expect(decodeLenBytes(buf, 0)).toBe(2);
+  });
+  test('encoded length of 3', () => {
+    const buf = Uint8Array.from(new Array(2));
+    // 130 signifies a DER length of 3
+    buf[0] = 130;
+    expect(decodeLenBytes(buf, 0)).toBe(3);
+  });
+  test('encoded length of 4', () => {
+    const buf = Uint8Array.from(new Array(2));
+    // 131 signifies a DER length of 4
+    buf[0] = 131;
+    expect(decodeLenBytes(buf, 0)).toBe(4);
+  });
+  test('encoded length of 4', () => {
+    const buf = Uint8Array.from(new Array(2));
+    buf[0] = 132;
+    const shouldFail = () => decodeLenBytes(buf, 0);
+    expect(shouldFail).toThrowError('Length too long');
+  });
+});
+
+describe('decodeLen', () => {
+  test('encoded length of 1', () => {
+    const buf = Uint8Array.of(1);
+    expect(decodeLen(buf, 0)).toBe(1);
+  });
+  test('encoded length of 2', () => {
+    const buf = Uint8Array.from(new Array(10));
+    buf[0] = 129;
+    // returns value stored at index 1
+    buf[1] = 5;
+
+    expect(decodeLen(buf, 0)).toBe(5);
+  });
+  test('encoded length of 3', () => {
+    const buf = Uint8Array.from(new Array(10));
+    buf[0] = 130;
+    // returns value stored at index 2
+    buf[2] = 5;
+    expect(decodeLen(buf, 0)).toBe(5);
```

Comment:
To test the addition that happens here, maybe switch to the following and remove the comment? 
```suggestion
    buf[1] = 1;
    buf[2] = 1;
    expect(decodeLen(buf, 0)).toBe(257);
```

Reply:
That is not a valid case 
![image](https://user-images.githubusercontent.com/16298804/130661658-40610ec2-c7fe-44e8-8fba-a427816eb578.png)

Reply:
I think you are just missing the initialization at position 0. This test is successful for me: 
```
  test('encoded length of 3', () => {
    const buf = Uint8Array.from(new Array(10));
    buf[0] = 130;
    buf[1] = 1;
    buf[2] = 1;
    expect(decodeLen(buf, 0)).toBe(257);
  });
```